### DISCUSSION TITLE
output.hls streams_info video size isn't optional

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#2176] [User Support]
**Describe the bug**
Using streams_info in output.hls the video size arg is not optional

**To Reproduce**
```
streams_info=[("live", (220000, "mp4a.40.40.2", "ts"))]
``` 
errors asking for ```[(_,(_,_,_,_))]```

```
streams_info=[("live", (220000, "mp4a.40.40.2", "ts", (1200,720)))]
```
works fine.

**Expected behavior**
streams_info to accept args without video size

**Version details**
 - OS: [e.g. Debian, OSX] Ubuntu 20.04
 - Version [e.g. `1.3.4`] 2.1.0

**Install method**
Opam

**Comment by toots** [comment]
So, this is not really a bug but a limitation of the type system. Currently, we have no way to make this argument optional without having to explicitly pass `null()`

I have just changed the argument to be a record so that now, it should be passed as:
```ruby
streams_info=[(
  "live", {
    bandwidth  = "220000",
    codecs     = "mp4a.40.40.2",
    extname    = "ts",
    video_size = null()
  }
)]
```

**Reply by Russsgithub** [reply]
you set bandwidth type to int not string ..... works fine.

Thanks.
